### PR TITLE
Add custom-property-set test to selector-type-no-unknown

### LINKED ISSUE #1791: False positives with CSS Custom Properties in selector-type-no-unknown
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug where `selector-type-no-unknown` doesn't understand CSS Custom Property syntax by default. 

> Which rule, if any, is this issue related to?

`selector-type-no-unknown`

> What CSS is needed to reproduce this issue?

```css
:root {
  --ThemeAnimationStyles {
    transition-duration: .2s;
    transition-timing-function: cubic-bezier(.075, .82, .165, 1);
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "number-leading-zero": "never",
    "custom-property-no-outside-root": true,
    "root-no-standard-properties": true,
    "selector-root-no-composition": true,
    "declaration-block-no-duplicate-properties": true,
    "no-eol-whitespace": true
  }
}
```

> Which version of stylelint are you using? 

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Maybe? 

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
✖  Unexpected unknown type selector "--ThemeAnimationStyles"
```

> Workaround 

```json
"selector-type-no-unknown": [true, {
    	"ignoreTypes": ["/^--/", "--Theme"]
  }]
```

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #1791

My bad, I totally didn’t see the missing semicolon in code.

### CODE REVIEW

**Comment by ntwb** [comment]
It's fantastic to see AppVeyor finishing before Travis CI 🎉

**Comment by ntwb** [comment]
LGTM

File: src/rules/selector-type-no-unknown/__tests__/index.js [hunk 1]
```diff
@@ -62,6 +62,9 @@ testRule(rule, {
   }, {
     code: "@include keyframes(identifier) { 0% {} 100% {} }",
     description: "non-standard usage of keyframe selectors",
+  }, {
+    code: "a { --custom-property-set: {} }",
```
Comment (davidtheclark) [inline src/rules/selector-type-no-unknown/__tests__/index.js]: I think it's supposed to have a semicolon after the `{}`, so `a { --custom-property-set: {}; }`

**Comment by davidtheclark** [comment]
One nitpicky detail then looks good to merge.

File: src/rules/selector-type-no-unknown/__tests__/index.js (hunk 1 above)
Comment (jeddy3) [inline src/rules/selector-type-no-unknown/__tests__/index.js]: Whoops, done.

**Pull request merged**